android/src/main/java/com/tailscale/ipn/mdm/MDMSettingsDefinitions.kt

```diff
@@ -4,14 +4,32 @@
 package com.tailscale.ipn.mdm
 
 import android.os.Bundle
+import android.util.Log
 import com.tailscale.ipn.App
 import com.tailscale.ipn.ui.util.set
 import kotlinx.coroutines.flow.MutableStateFlow
 import kotlinx.coroutines.flow.StateFlow
 
-abstract class MDMSetting<T>(defaultValue: T, val key: String, val localizedTitle: String) {
+// Don't keep track of the key if shouldRegisterKey is false - shouldRegisterKey should be false when creating an MDMSetting for retrieving a setting.
+abstract class MDMSetting<T>(defaultValue: T, val key: String, val localizedTitle: String, private val shouldRegisterKey: Boolean = true) {
+  private val keys = mutableSetOf<String>()
+
+    init {
+      if (shouldRegisterKey) { registerKey(key)
+    }
+  }
   val flow: StateFlow<T> = MutableStateFlow<T>(defaultValue)
 
+  private fun registerKey(key: String) {
+    if (!keyExists(key)) {
+        keys.add(key)
+    }
+}
+
+fun keyExists(key: String): Boolean {
```

Comment:
nit: the one line method here isn't really needed (unless it's called by something else I'm not seeing.)

Reply:
we are using this in App.getSyspolicyStringValue - MDMSettings has all the MDMSettings we've created, and we want to check to see if there is one with the key we're given